File: src/pub_sub/producer.ts
import {connect} from "amqplib";

const producer = async () => {
    const connection = await connect({
        hostname: process.env.AMQP_HOSTNAME,
        port: parseInt(process.env.AMQP_PORT as string) || 5672,
        username: process.env.AMQP_USERNAME,
        password: process.env.AMQP_PASSWORD
    });

    const channel = await connection.createChannel();
    await channel.assertExchange("pubsub", "fanout");
    channel.publish("pubsub", "", Buffer.from("This message should be broadcasting."));
    await channel.close();
    await connection.close();
}

export default producer;

File: cmd/pub_sub/producer.go
package pub_sub

import (
	"rabbitmq/utils"

	"github.com/spf13/cobra"
	"github.com/streadway/amqp"
)

var Producer = &cobra.Command{
	Use:   "pub_sub_producer",
	Short: "Publish message to message queue.",
	Run: func(cmd *cobra.Command, args []string) {
		conn, err := amqp.Dial(utils.Env.AMQP_URL)
		utils.LogError(err, "Can not connect to amqp message server")
		defer conn.Close()

		ch, err := conn.Channel()
		utils.LogError(err, "Can not create a channel")
		defer ch.Close()

		err = ch.ExchangeDeclare("pub_sub", amqp.ExchangeFanout, false, true, false, false, nil)
		utils.LogError(err, "Can not declare a exchange.")

		err = ch.Publish("pub_sub", "", false, false, amqp.Publishing{Body: []byte("This message should broadcasting.")})
		utils.LogError(err, "Can not publish message to queue")
	},
}
